implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -101,17 +111,21 @@ private void createDatabaseIfNecessary() {
 
     @Override
     protected void publish() {
+        recognizeInfluxDBVersion();
         createDatabaseIfNecessary();
 
         try {
-            String influxEndpoint = config.uri() + "/write?consistency=" + config.consistency().toString().toLowerCase() + "&precision=ms&db=" + config.db();
+            String influxEndpoint = influxDBVersion.writeEndpoint(this);
             if (StringUtils.isNotBlank(config.retentionPolicy())) {
                 influxEndpoint += "&rp=" + config.retentionPolicy();
```

Comment:
While not harmful (the write request is still successful with this), `rp` is not a parameter that the v2 write endpoint recognizes.

Reply:
Fixed